crates/sui-graphql-rpc/src/server/data_provider.rs

```diff
@@ -216,6 +221,108 @@ fn convert_bal(b: sui_json_rpc_types::Balance) -> Balance {
         total_balance: BigInt::from_str(&format!("{}", b.total_balance)).unwrap(),
     }
 }
+pub(crate) async fn fetch_gas_input(cl: &SuiClient, digest: &String) -> Result<GasInput> {
+    let tx_block = cl
+        .read_api()
+        .get_transaction_with_options(
+            TransactionDigest::from_str(digest)?,
+            SuiTransactionBlockResponseOptions::full_content(),
```

Comment:
This is not strictly necessary -- you only need to see the transaction itself, i.e. `show_input`.